### PR TITLE
Add full component state/command and snapshot tests

### LINKED ISSUE #191: Add full component state coverage and snapshot tests
## Summary

Add comprehensive test coverage for all components across stdlib, textarea, and treeview crates, covering both state/command tests and render snapshot tests.

### 1. State and command tests (full coverage)

For every component in stdlib, textarea, and treeview:

- Test all `perform(Cmd) -> CmdResult` transitions
- Assert on `state()` after each command
- Cover edge cases (empty state, boundary values, wrapping behavior, etc.)

### 2. Render snapshot tests

Using the `TestTerminalAdapter` (already implemented):

- Render each component with representative props and assert on the buffer contents
- Use snapshot-style assertions to catch visual regressions
- Cover key visual states (focused/unfocused, different content lengths, etc.)

### PR DESCRIPTION
**Pull request by veeso**
## Summary

Closes #191

- Add comprehensive state/command test coverage for all 21 components across stdlib (19), textarea (1), and treeview (1) crates
- Add render snapshot tests using `insta` + `TestTerminalAdapter` for visual regression detection
- Shared test helpers (`render_to_string`, `buffer_to_string`) in each crate's `tests/common.rs`

## What's tested

| Crate | Components | Tests |
|-------|-----------|-------|
| stdlib | BarChart, Canvas, Chart, Checkbox, Container, Gauge, Input, Label, LineGauge, List, Paragraph, Phantom, Radio, Select, Span, Sparkline, Spinner, Table, Textarea | ~95 |
| textarea | TextArea | 21 |
| treeview | TreeView | 10 |

**Per component:**
- All `perform(Cmd) -> CmdResult` transitions
- `state()` assertions after each command
- Edge cases (empty, boundary, disabled, rewind)
- Render snapshots (default, focused, content variations)

## Test plan

- [x] `cargo test --workspace --all-features` — all pass
- [x] `cargo clippy --workspace --all-targets --all-features -- -Dwarnings` — clean
- [x] `cargo +nightly fmt --all -- --check` — clean

### CODE REVIEW

File: crates/tuirealm-stdlib/tests/common.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    result
+}
+
+/// Render a component into a TestTerminalAdapter and return the buffer as a string.
+pub fn render_to_string(component: &mut dyn Component, width: u16, height: u16) -> String {
+    let mut adapter = TestTerminalAdapter::new(width, height).unwrap();
+    let area = Rect::new(0, 0, width, height);
+    let completed = adapter.raw_mut().draw(|f| component.view(f, area)).unwrap();
+    buffer_to_string(completed.buffer)
+}
```
Comment (hasezoey) [inline crates/tuirealm-stdlib/tests/common.rs]: Instead of having this in each crate and user's projects, shouldnt we just de-duplicate it into core?

For Example, export all testing utils under a `tuirealm::testing` module?

File: crates/tuirealm-textarea/tests/component_textarea.rs [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+        CmdResult::None
+    );
+}
+
+#[test]
+fn test_textarea_undo_redo() {
+    let mut component = TextArea::default();
+    component.perform(Cmd::Type('A'));
+    let result = component.perform(Cmd::Custom(TEXTAREA_CMD_UNDO));
+    assert!(matches!(result, CmdResult::Changed(_) | CmdResult::None));
```
Comment (hasezoey) [inline crates/tuirealm-textarea/tests/component_textarea.rs]: Shouldnt this be just one Result to be matched against?

File: crates/tuirealm-textarea/tests/component_textarea.rs [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+    component.perform(Cmd::Custom(TEXTAREA_CMD_REDO));
```
Comment (hasezoey) [inline crates/tuirealm-textarea/tests/component_textarea.rs]: Shouldnt this return also be tested?

File: crates/tuirealm-textarea/tests/component_textarea.rs
Comment (hasezoey) [inline crates/tuirealm-textarea/tests/component_textarea.rs:1]: From what i could tell, `::paste` is not in the tests yet.

File: crates/tuirealm-textarea/tests/component_textarea.rs [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+
+#[test]
+fn test_textarea_snapshot_default() {
+    let mut component = TextArea::default()
+        .borders(Borders::default())
+        .title(Title::from("Editor"))
+        .style(Style::default().fg(Color::White));
+    for ch in "Hello, World!".chars() {
+        component.perform(Cmd::Type(ch));
+    }
```
Comment (hasezoey) [inline crates/tuirealm-textarea/tests/component_textarea.rs:275]: Shouldnt this test be called `singleline`?

File: crates/tuirealm-stdlib/tests/component_phantom.rs [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+fn test_phantom_state_is_none() {
+    let component = Phantom::default();
+    assert_eq!(component.state(), State::None);
+}
+
+#[test]
+fn test_phantom_snapshot_default() {
+    let mut component = Phantom::default();
+    let rendered = common::render_to_string(&mut component, 40, 5);
+    insta::assert_snapshot!("phantom_default", rendered);
```
Comment (hasezoey) [inline crates/tuirealm-stdlib/tests/component_phantom.rs]: This could likely be a inline snapshot as nothing actually gets rendered.

File: crates/tuirealm-stdlib/tests/component_spinner.rs [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    assert_eq!(component.perform(Cmd::Delete), CmdResult::None);
+}
+
+#[test]
+fn test_spinner_snapshot_default() {
+    let mut component = Spinner::default()
+        .foreground(Color::Cyan)
+        .sequence("⠋⠙⠹⠸⠼⠴⠦⠧⠇⠏");
+    let rendered = common::render_to_string(&mut component, 10, 1);
+    insta::assert_snapshot!("spinner_default", rendered);
```
Comment (hasezoey) [inline crates/tuirealm-stdlib/tests/component_spinner.rs]: As this is also just one single character rendered, this could be a inline snapshot.

File: crates/tuirealm-stdlib/tests/snapshots/component_gauge__gauge_full.snap [hunk 7]
```diff
@@ -0,0 +1,7 @@
+---
+source: crates/tuirealm-stdlib/tests/component_gauge.rs
+expression: rendered
+---
+┌──────────────────────────────────────┐
+│███████████████████ ██████████████████│
```
Comment (hasezoey) [inline crates/tuirealm-stdlib/tests/snapshots/component_gauge__gauge_full.snap]: This looks like a bug we should fix: if there is no label, dont apply it. (or if it is added in "default", we shouldnt)

**Review by hasezoey — APPROVED** [review]
Looks great! Though not *everything* is tested, this is a major improvement.
Some minor questions regarding style:
- I dont quite like that some tests are called `default` even though they include more than `::default()`, how about something like `simple`? (though i know that `_empty` practically tests *just* `::default()`)
- insta snapshots can get the name from the calling function, in my opinion we should use that instead of custom names to easily identify the function it is called from as we practically (currently) only have one snapshot per function

**Comment by hasezoey** [comment]
> [chore: Do not track Cargo.lock](https://github.com/veeso/tui-realm/pull/197/commits/88b3a18e4392ef54b2e6998669ee9449f0174fd1)

Why exclude the Cargo.lock again? This helps us have consistent dependency version between us and the CI. (though i also understand the case of a new version breaking things and us not noticing until update or user reports it)
